Jetcaster/mobile/src/main/java/com/example/jetcaster/ui/home/category/PodcastCategory.kt

```diff
@@ -58,26 +63,44 @@ fun LazyListScope.podcastCategory(
     podcastCategoryFilterResult: PodcastCategoryFilterResult,
     navigateToPodcastDetails: (PodcastInfo) -> Unit,
     navigateToPlayer: (EpisodeInfo) -> Unit,
+    removeFromQueue: (EpisodeInfo) -> Unit,
     onQueueEpisode: (PlayerEpisode) -> Unit,
     onTogglePodcastFollowed: (PodcastInfo) -> Unit,
 ) {
     item {
         CategoryPodcasts(
             topPodcasts = podcastCategoryFilterResult.topPodcasts,
             navigateToPodcastDetails = navigateToPodcastDetails,
-            onTogglePodcastFollowed = onTogglePodcastFollowed
+            onTogglePodcastFollowed = onTogglePodcastFollowed,
         )
     }
 
     val episodes = podcastCategoryFilterResult.episodes
     items(episodes, key = { it.episode.uri }) { item ->
-        EpisodeListItem(
-            episode = item.episode,
-            podcast = item.podcast,
-            onClick = navigateToPlayer,
-            onQueueEpisode = onQueueEpisode,
-            modifier = Modifier.fillParentMaxWidth()
-        )
+        val sharedTransitionScope = LocalSharedTransitionScope.current
+            ?: throw IllegalStateException("No SharedElementScope found")
+        val animatedVisibilityScope = LocalAnimatedVisibilityScope.current
+            ?: throw IllegalStateException("No SharedElementScope found")
```

Comment:
You may want to do this a bit differently for this example since the Large screen versions wont need to have the scopes provided, you can probably build up a version that rather allows this to be nullable. 

We've explored a few different ways to do this - the least amount of code would probably be this kind of block:
```
.block {
    val animatedVisibilityScope = LocalNavAnimatedVisibilityScope.current
    if(this@with != null && animatedVisibilityScope != null)
        sharedElement(
            this@with.rememberSharedContentState(
                key = ChatSharedElementKey(
                    chatId = chat.chatWithLastMessage.id,
                    type = ChatSharedElementType.Image)
            ),
            animatedVisibilityScope = animatedVisibilityScope
        ) else Modifier
}
```
I also explored it with Jetsnack in creating a whole bunch of modifiers that are called `safe` versions of things. 
https://github.com/android/compose-samples/pull/1417/files#diff-db028e384f27d7d2dd08354e752447d4efddb1faaa8dafc3f43e7c73682e0270